### PR TITLE
Close #12087: Use ScreenCoordsXY on gfx_filter_pixel

### LINKED ISSUE #12087: Use ScreenCoordsXY on gfx_filter_pixel
- Create an overload of `gfx_filter_pixel` with the same name, but that takes a `const ScreenCoordsXY&` object instead of an `int32_t x` and and an `int32_t y` 
- Make the old function call the new one to avoid code duplication
- Replace the calls to the `{x ,y}` pair by the one using `ScreenCoordsXY`.
- If no calls to the old one are left, remove it.

*Note*: Don't try to update all calls at once, unless it doesn't occur a lot, as this might lead to a huge PR.

### PR DESCRIPTION
**Pull request by annoyingdroid**
This is my first pull request, so my apologies for any errors. 

Modified `gfx_filter_pixel` to now use `const ScreenCoordsXY&`, and changed calls to the old function to reference the new one (there are only 4 calls to the function).

### CODE REVIEW

**Comment by annoyingdroid** [comment]
Got it. With the rebase, is it necessary that I force the push after rebasing? I'm running into an issue where the push fails because the branch tip is behind the remote counterpart.

**Comment by tupaschoal** [comment]
Yes

File: src/openrct2-ui/windows/Tooltip.cpp [hunk 1]
```diff
@@ -203,10 +203,10 @@ static void window_tooltip_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
 
     // Corners
-    gfx_filter_pixel(dpi, left + 1, top + 1, PALETTE_DARKEN_3);
-    gfx_filter_pixel(dpi, right - 1, top + 1, PALETTE_DARKEN_3);
-    gfx_filter_pixel(dpi, left + 1, bottom - 1, PALETTE_DARKEN_3);
-    gfx_filter_pixel(dpi, right - 1, bottom - 1, PALETTE_DARKEN_3);
+    gfx_filter_pixel(dpi, ScreenCoordsXY{ left + 1, top + 1 }, PALETTE_DARKEN_3);
+    gfx_filter_pixel(dpi, ScreenCoordsXY{ right - 1, top + 1 }, PALETTE_DARKEN_3);
+    gfx_filter_pixel(dpi, ScreenCoordsXY{ left + 1, bottom - 1 }, PALETTE_DARKEN_3);
+    gfx_filter_pixel(dpi, ScreenCoordsXY{ right - 1, bottom - 1 }, PALETTE_DARKEN_3);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Tooltip.cpp:209]:
```suggestion
    gfx_filter_pixel(dpi, { left + 1, top + 1 }, PALETTE_DARKEN_3);
    gfx_filter_pixel(dpi, { right - 1, top + 1 }, PALETTE_DARKEN_3);
    gfx_filter_pixel(dpi, { left + 1, bottom - 1 }, PALETTE_DARKEN_3);
    gfx_filter_pixel(dpi, { right - 1, bottom - 1 }, PALETTE_DARKEN_3);
```
Compiler can infer the type, so we usually do it this way for shortness

Comment (annoyingdroid) [reply src/openrct2-ui/windows/Tooltip.cpp:209]: Good to know! Let me make those changes.

**Pull request merged**